README.md
yum-remi Cookbook
=================

The yum-remi cookbook takes over management of the repositoryids of
the [remi](http://rpms.remirepo.net/) repository . It allows
attribute manipulation of `remi`, `remi-php55`, `remi-php56`,
`remi-php70`, and `remi-test` repositories.

Requirements
------------
* Chef 11 or higher
* yum cookbook version 3.0.0 or higher

Platform Support
----------------
The following platforms have been tested with Test Kitchen:

```
|-----------+------+------------+------------+------------|
|           | remi | remi-php55 | remi-php56 | remi-php70 |
|-----------+------+------------+------------+------------|
| centos-5  | X    | X          | X          |            |
|-----------+------+------------+------------+------------|
| centos-6  | X    | X          | X          | X          |
|-----------+------+------------+------------+------------|
| centos-7  | X    | X          | X          | X          |
|-----------+------+------------+------------+------------|
| redhat-7  | X    | X          | X          | X          |
|-----------+------+------------+------------+------------|
| fedora-21 | X    |            |            | X          |
|-----------+------+------------+------------+------------|
| fedora-22 | X    |            |            | X          |
|-----------+------+------------+------------+------------|
| fedora-23 | X    |            |            | X          |
|-----------+------+------------+------------+------------|
```

Amazon Linux is *not* supported by the Remi repository. Amazon
maintains their own PHP packages natively, as php53, php54, php55, and
php56.

Attributes
----------
The following attributes are set by default

``` ruby
default['yum']['remi']['repositoryid'] = 'remi'
default['yum']['remi']['mirrorlist'] = 'http://rpms.remirepo.net/enterprise/6/remi/mirror'
default['yum']['remi']['description'] = "Remi's RPM repository for Enterprise Linux 6 - $basearch"
default['yum']['remi']['enabled'] = true
default['yum']['remi']['gpgcheck'] = true
default['yum']['remi']['gpgkey'] = 'http://rpms.remirepo.net/RPM-GPG-KEY-remi'
```

``` ruby
default['yum']['remi-php55']['repositoryid'] = 'remi-php55'
default['yum']['remi-php55']['mirrorlist'] = 'http://rpms.remirepo.net/enterprise/6/php55/mirror'
default['yum']['remi-php55']['description'] = "Remi's PHP 5.5 RPM repository for Enterprise Linux 6 - $basearch"
default['yum']['remi-php55']['enabled'] = true
default['yum']['remi-php55']['gpgcheck'] = true
default['yum']['remi-php55']['gpgkey'] = 'http://rpms.remirepo.net/RPM-GPG-KEY-remi'
```

``` ruby
default['yum']['remi-php55']['repositoryid'] = 'remi-php56'
default['yum']['remi-php55']['mirrorlist'] = 'http://rpms.remirepo.net/enterprise/6/php56/mirror'
default['yum']['remi-php55']['description'] = "Remi's PHP 5.6 RPM repository for Enterprise Linux 6 - $basearch"
default['yum']['remi-php55']['enabled'] = true
default['yum']['remi-php55']['gpgcheck'] = true
default['yum']['remi-php55']['gpgkey'] = 'http://rpms.remirepo.net/RPM-GPG-KEY-remi'
```

``` ruby
default['yum']['remi-php55']['repositoryid'] = 'remi-php70'
default['yum']['remi-php55']['mirrorlist'] = 'http://rpms.remirepo.net/enterprise/6/php70/mirror'
default['yum']['remi-php55']['description'] = "Remi's PHP 7.0 RPM repository for Enterprise Linux 6 - $basearch"
default['yum']['remi-php55']['enabled'] = true
default['yum']['remi-php55']['gpgcheck'] = true
default['yum']['remi-php55']['gpgkey'] = 'http://rpms.remirepo.net/RPM-GPG-KEY-remi'
```

Recipes
-------
* default - Walks through node attributes and feeds a yum_resource
  parameters. The following is an example a resource generated by the
  recipe during compilation.

```ruby
  yum_repository 'remi' do
    mirrorlist 'http://rpms.remirepo.net/enterprise/6/remi/mirror'
    description "Remi's RPM repository for Enterprise Linux 6 - $basearch"
    enabled true
    gpgcheck true
    gpgkey 'http://rpms.remirepo.net/RPM-GPG-KEY-remi'
  end
```

Usage Example
-------------
To disable the remi repository through a Role or Environment definition

```
default_attributes(
  :yum => {
    :remi => {
      :enabled => {
        false
       }
     }
   }
 )
```

More Examples
-------------
Point the base and updates repositories at an internally hosted server.

```
node.default['yum']['remi']['enabled'] = true
node.default['yum']['remi']['mirrorlist'] = nil
node.default['yum']['remi']['baseurl'] = 'https://internal.example.com/enterprise/5/remi/$basearch/'
node.default['yum']['remi']['sslverify'] = false

include_recipe 'yum-remi'
```

License & Authors
-----------------
Authors:
- Lucas Mariani - Psyreactor [Lead] (<marianiluca@gmail.com>)
    - https://github.com/psyreactor
    - http://blog.psyreactor.com.ar
- Sean OMeara (<sean@chef.io>)
- Andrew Miller (<andrew.miller@rakuten.com>)


CHANGELOG.md
yum-remi Cookbook CHANGELOG
===========================

v1.0.0
------
Removed support for Fedora linux older than 21.
Updated for most recent remi repositories.
Added remi-php70.

v0.1.0
------
initial release


attributes/remi-dev.rb
default['yum']['remi-dev']['repositoryid'] = 'remi-dev'
default['yum']['remi-dev']['gpgkey'] = 'http://rpms.remirepo.net/RPM-GPG-KEY-remi'
default['yum']['remi-dev']['gpgcheck'] = true
default['yum']['remi-dev']['enabled'] = false
default['yum']['remi-dev']['managed'] = false

case node['platform']
when 'fedora'
  case node['platform_version'].to_i
  when 21
    default['yum']['remi-dev']['baseurl'] = 'http://rpms.remirepo.net/fedora/21/devs/$basearch/'
    default['yum']['remi-dev']['description'] = "Remi's development RPM repository for Fedora Linux 21 - $basearch"
  when 22
    default['yum']['remi-dev']['baseurl'] = 'http://rpms.remirepo.net/fedora/22/devs/$basearch/'
    default['yum']['remi-dev']['description'] = "Remi's development RPM repository for Fedora Linux 22 - $basearch"
  when 23
    default['yum']['remi-dev']['baseurl'] = 'http://rpms.remirepo.net/fedora/20/devs/$basearch/'
    default['yum']['remi-dev']['description'] = "Remi's development RPM repository for Fedora Linux 23 - $basearch"
  end
else
  case node['platform_version'].to_i
  when 5
    default['yum']['remi-dev']['baseurl'] = 'http://rpms.remirepo.net/enterprise/5/devs/$basearch/'
    default['yum']['remi-dev']['description'] = "Remi's development RPM repository for Enterprise Linux 5 - $basearch"
  when 6
    default['yum']['remi-dev']['baseurl'] = 'http://rpms.remirepo.net/enterprise/6/devs/$basearch/'
    default['yum']['remi-dev']['description'] = "Remi's development RPM repository for Enterprise Linux 6 - $basearch"
  when 7
    default['yum']['remi-dev']['baseurl'] = 'http://rpms.remirepo.net/enterprise/7/devs/$basearch/'
    default['yum']['remi-dev']['description'] = "Remi's development RPM repository for Enterprise Linux 7 - $basearch"
  end
end
